### PR TITLE
Commited in the 47154_user_creation

### PR DESCRIPTION
**Pull request by mdmahmudurrahman**

### CODE REVIEW

File: app/controllers/registrations_controller.rb [hunk 1]
```diff
@@ -1,6 +1,9 @@
 class RegistrationsController < Devise::RegistrationsController
   private
   def sign_up_params
```
Comment (shakil-fg) [inline app/controllers/registrations_controller.rb]: Add additional indentation for every method after private, like this:

private:
  def account_update_params
    ...
  end

File: app/views/layouts/_header.html.erb [hunk 2]
```diff
@@ -34,6 +34,16 @@
                 <input type="password" placeholder= <%= t(:password) %> class="form-control">
               </div>
               <button type="submit" class="btn btn-success"><% t(:sign_in)%>></button>
+              <% if !user_signed_in? %>
+              <div class="form-group">
```
Comment (shakil-fg) [inline app/views/layouts/_header.html.erb]: indentation

File: db/seeds.rb [hunk 3]
```diff
@@ -1,37 +1,37 @@
-Subject.create(name: 'Algorithm', description: 'Heart of computer science, We read book of Core Man.', tasks_attributes: [{name: 'Computation complexity', description: 'Analysis of performance of an algorithm'}, {name: 'Greeedy algorithm', description: 'Algorithm finding the best things first'}, {name: 'Divide and conquare', description: 'Solving problem breaking it into smaller part'}, {name: 'Searching algorithm', description: 'Binary search'}, {name: 'Sorting algorithm', description: 'Bubble, quick, merge, insertion, selection, heap sort'}])
```
Comment (shakil-fg) [inline db/seeds.rb]: repair seed file

File: app/views/layouts/_header.html.erb [hunk 4]
```diff
@@ -34,6 +34,16 @@
...
+                 <%= link_to t(:sign_in_text), new_user_session_path %>
+              </div>
+              <% else %>
+              <div class="form-group">
+                <%= link_to t(:sign_out_text), destroy_user_session_path %>
+              </div>
+              <% end %>
+
```
Comment (noor-framgia) [inline app/views/layouts/_header.html.erb]: remove empty line

File: app/views/devise/registrations/new.html.erb [hunk 5]
```diff
@@ -25,9 +25,13 @@
     <%= f.label :password_confirmation %><br />
     <%= f.password_field :password_confirmation, autocomplete: "off" %>
   </div>
-
+    <div class="field">
+      <%= f.label :role %><br />
+      <%= f.select(:role, options_for_select([["Supervisor", "Supervisor" ],["Trainee", "Trainee"]]), prompt:"Select role") %>
```
Comment (chuanhtuan) [inline app/views/devise/registrations/new.html.erb]: define constant in user model

**Pull request merged**